### PR TITLE
Ncnews 3

### PR DESCRIPTION
**Pull request by radzarom**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+    return db.end();
+});
 
 
 describe('app.js test suite', () => {
+
     describe('GET /api/topics', () => {
-        test('should ', () => {
-            
+        test('responds with status code 200', () => {
```
Comment (mezzDavies) [inline __tests__/app.test.js:20]: This assertion is implicit in the following test (you expect a 200 status there too) so doesn't need to be a standalone test. Should save you some time further down the line 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+            return request(app)
+            .get('/api/topics')
+            .expect(200)
         });
+
+        test('responds with an array of topic objects with slug and description properties', () => {
+            return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
```
Comment (mezzDavies) [inline __tests__/app.test.js:30]: There should be an assertion that the length of the array is what you should expect it to be.

File: db/controller.js [hunk 3]
```diff
@@ -1 +1,15 @@
...
+
+
+const getTopics = (request, response, next) => {
+
+    retrieveTopics().then((topics) => {
+
+        response.status(200).send({topics: topics});
+    })
+    .catch((err) => {
+        next(err);
```
Comment (mezzDavies) [inline db/controller.js:11]: 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -1,10 +1,43 @@
...
+                body.topics.forEach((topic) => {
+                    expect(topic).toHaveProperty('slug', expect.any(String));
+                    expect(topic).toHaveProperty('description', expect.any(String));
+                })
+
+            })
+        });
+
+        // test('responds with 404 if ', () => {
```
Comment (mezzDavies) [inline __tests__/app.test.js:39]: Your instincts are correct You should finish this test

File: db/app.js [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
 const app = express()
-app.use(express.json());
\ No newline at end of file
+app.use(express.json());
+
+app.get('/api/topics', getTopics)
+
+app.use((error, request, response, next) => {
+
+    response.status(404).send({message: 'Something went wrong'})
```
Comment (mezzDavies) [inline db/app.js:11]: Please complete the testing for this. Remember though TDD is test DRIVEN development 😄

**Review by mezzDavies — APPROVED** [review]

File: __tests__/app.test.js (hunk 4 above)
Comment (radzarom) [reply __tests__/app.test.js:39]: Hi, I'm not sure what to put in a test here. If I do request(app).get('/api/topics') I don't see anyway I can force that to error like I could with using an invalid :id if you know what I mean?

**Pull request merged**